### PR TITLE
Explain prior responses issue for App Interceptors

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
For discussion, tests to come.

File: okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt [hunk 1]
```diff
@@ -122,7 +122,7 @@ class Transmitter(
       if (this.request!!.url.canReuseConnectionFor(request.url) && exchangeFinder!!.hasRouteToTry()) {
         return // Already ready.
       }
-      check(exchange == null)
+      check(exchange == null) { "Transmitted has existing exchange, has an app interceptor left prior responses open?" }
```
Comment (octylFractal) [inline okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt]:
```suggestion
      check(exchange == null) { "Transmitter has existing exchange, has an app interceptor left prior responses open?" }
```
I assume this is what was intended.

Comment (yschimke) [reply okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt]: Thanks

File: okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt [hunk 2]
```diff
@@ -122,7 +122,7 @@ class Transmitter(
       if (this.request!!.url.canReuseConnectionFor(request.url) && exchangeFinder!!.hasRouteToTry()) {
         return // Already ready.
       }
-      check(exchange == null)
+      check(exchange == null) { "Transmitter has existing exchange, has an app interceptor left prior responses open?" }
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt]: Wanna use the exact same text we use on line 158?

Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt]: (possibly fixing that line to match this!)

**Pull request merged**